packages/wire/lib/gossip/DnsPeerQuery.ts

```diff
@@ -0,0 +1,116 @@
+import { ILogger } from "@node-lightning/logger";
+import { PeerHostRecord } from "../PeerHostRecord";
+import { AnySrvRecord } from "dns";
+const { Resolver } = require("dns").promises;
+import bech32 from "bech32";
+
+export interface DnsPeerQueryOptions {
+    /**
+     * The domain of the dns seed.
+     */
+    dnsSeed: string;
+
+    /**
+     * The realm byte used to specify what realm the returned nodes must support.
+     */
+    r?: number;
+
+    /**
+     * The address type bit field specifies the address types that should be returned.
+     * It follows the format of the address descriptor type specified in BOLT #7.
+     */
+    a?: number;
+
+    /**
+     * The bech32-encoded node id used to retrive the result of a single node.
+     */
+    l?: string;
+
+    /**
+     * The number of desired reply records.
+     */
+    n?: number;
+}
+
+/**
+ * Performs a dns lookup on a dns seed to discover lightning nodes.
```

Comment:
Mind adding some comments on how the specification works?  It's useful for other developers to understand why the code is doing what its doing.